ecommerce/api.py

```diff
@@ -108,6 +114,37 @@ def generate_cybersource_sa_payload(order, base_url):
 
         total += unit_price
 
+    # At the moment there should only be one line
+    product_version = order.lines.first().product_version
+    product = product_version.product
+    content_object = product.content_object
+
+    merchant_fields = {
+        "merchant_defined_data1": str(product.content_type),
+        "merchant_defined_data2": (
+            (
+                content_object.readable_id
+                if isinstance(content_object, Program)
+                else content_object.courseware_id
```

Comment:
If `content_object.courseware_id` is not defined for a course run (maybe it should never be in production, but it's not a required field), then I get the following error from cybersource:

![Screen Shot 2019-06-24 at 10 06 43 AM](https://user-images.githubusercontent.com/187676/60027488-93fcdc00-966b-11e9-88fd-a1f284af30a9.png)

Setting "merchant_defined_data2" to an empty string instead of `None` works though.

Reply:
Ah good catch